control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -696,12 +698,22 @@ role_arn = %s
 web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
 `
 
-func (r *reconciler) reconcileCloudCredentialSecrets(ctx context.Context, hcp *hyperv1.HostedControlPlane) []error {
+func (r *reconciler) reconcileCloudCredentialSecrets(ctx context.Context, hcp *hyperv1.HostedControlPlane, log logr.Logger) []error {
 	var errs []error
 	switch hcp.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
 		for _, role := range hcp.Spec.Platform.AWS.Roles {
 			secret := manifests.AWSCloudCredsSecret(role)
+			ns := &corev1.Namespace{}
+			err := r.client.Get(ctx, client.ObjectKey{Name: secret.Namespace}, ns)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					log.Info("WARNING: cannot sync cloud credential secret because namespace does not exist", "secret", client.ObjectKeyFromObject(secret))
```

Comment:
When do we expect this to be the case? Does the CVO create that NS or who does?

Reply:
The cloud network controller doesn't exist in 4.9, causing every sync to fail if you're running a 4.9 release payload (https://github.com/openshift/hypershift/pull/804)